components/script/dom/canvasrenderingcontext2d.rs

```diff
@@ -1071,10 +1070,12 @@ impl CanvasRenderingContext2DMethods for CanvasRenderingContext2D {
 
         // Un-premultiply alpha
         for chunk in data.chunks_mut(4) {
-            let alpha = chunk[3] as usize;
-            chunk[0] = UNPREMULTIPLY_TABLE[256 * alpha + chunk[0] as usize];
-            chunk[1] = UNPREMULTIPLY_TABLE[256 * alpha + chunk[1] as usize];
-            chunk[2] = UNPREMULTIPLY_TABLE[256 * alpha + chunk[2] as usize];
+            let alpha = chunk[3] as u16;
+            if alpha != 0 {
```

Comment:
If `alpha` is `0`, then the chunk is full of zeros, and we leave those as zeros when unpremultiplying. Is that the desired behavior?